tasks/managed/update-package-collection-after-push/update-package-collection-after-push.yaml

```
@@ -0,0 +1,475 @@
+        - name: workDir
+          value: $(params.dataDir)
+        - name: sourceDataArtifact
+          value: $(params.sourceDataArtifact)
+    - name: update-package-collection-after-push
+      image: "$(params.pipelineImage)"
+      computeResources:
+        limits:
+          memory: 1G
+        requests:
+          memory: 1G
+          cpu: '1'
+      env:
+        - name: SNAPSHOT_SPEC_FILE
+          value: $(params.dataDir)/$(params.snapshotPath)
+        - name: DATA_FILE
+          value: $(params.dataDir)/$(params.dataPath)
+        - name: ACCESS_TOKEN
+          value: "xW6mhRVHcMxycV-vmLN7"
+        - name: GITLAB_HOST
+          value: "gitlab.cee.redhat.com"
+        - name: GIT_AUTHOR_NAME
+          value: "pkg_coll_auto_bot"
+        - name: GIT_AUTHOR_EMAIL
+          value: "pkg_coll_auto_bot@redhat.com"
+        - name: GIT_REPO
+          value: "https://gitlab.cee.redhat.com/package-collection/package-collection-utils.git"
+      volumeMounts:
+        - mountPath: /etc/secret
+          name: secret-volume
+          readOnly: true
+      # - name: package-collection-secret
+      #  mountPath: /mnt/package_collection_secret
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        # Set parameters as local variables
+        MAX_RETRIES="$(params.max_retries)"
+        COMMAND_TIMEOUT="$(params.command_timeout)"
+
+        # Source utility functions
+        source /home/utils/common-functions.sh 2>/dev/null || true
+
+        # Initialize environment
+        initialize_environment() {
+          validate_input_files
+          setup_environment_variables
+          setup_kerberos_auth
+          setup_git_repository
+        }
+
+        # Validate required input files
+        validate_input_files() {
+          if [ ! -f "${SNAPSHOT_SPEC_FILE}" ]; then
+            echo "ERROR: No valid snapshot file was provided at ${SNAPSHOT_SPEC_FILE}"
+            exit 1
+          fi
+
+          if [ ! -f "${DATA_FILE}" ]; then
+            echo "ERROR: No data JSON was provided at ${DATA_FILE}"
+            exit 1
+          fi
+        }
+
+        # Setup environment variables from JSON files
+        setup_environment_variables() {
+          export APPLICATION=$(jq -r '.application' "${SNAPSHOT_SPEC_FILE}")
+          export NUM_COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
+          export PRODUCT_SHORT_NAME=$(jq -re '.releaseNotes.product_short_name' "$DATA_FILE")
+          export INTENTION=$(jq -r '.intention' "$DATA_FILE")
+          export PRINCIPAL=$(jq -r '.pushOptions.pushKeytab.principal' "${DATA_FILE}")
+          export KEYTAB_FILE=$(jq -r '.pushOptions.pushKeytab.name' "${DATA_FILE}")
+          export KOJI_PROFILE=$(jq -r '.pushOptions.koji_profile' "${DATA_FILE}")
+          export KOJI_TAGS=$(jq -r '.pushOptions.koji_tags // [] | join(" ")' "${DATA_FILE}")
+          export KOJI_IMPORT_DRAFT=$(jq -r '.pushOptions.koji_import_draft' "${DATA_FILE}")
```

```suggestion
          export KOJI_TAGS=$(jq -r '.pushOptions.koji_tags // [] | join(" ")' "${DATA_FILE}")
          export KOJI_IMPORT_DRAFT=$(jq -r '.pushOptions.koji_import_draft' "${DATA_FILE}")
```
Get the promote value inside the pipeline and based the pipeline to set or not set these two values
       - name: promoteBuild
          value: $(tasks.collect-task-params.results.extractedValues[3])
          
 Add the 
          # Extract username from principal
          export USER_NAME=$(echo "$PRINCIPAL" | cut -d'@' -f1)
           # Extract username from principal
           export USER_NAME=$(echo "$PRINCIPAL" | cut -d'@' -f1)
          
 when the promote is true.
 
Plus another comment I left, I feel like there is no much diff between these two tasks. I feel like we can merge them to one and we call two times with different value of the promoteBuild in the push-rpm-to-koji pipeline, what do you think?